packages/core/core/src/RequestTracker.js

```diff
@@ -384,32 +485,85 @@ export class RequestGraph extends Graph<
       .filter(Boolean);
   }
 
+  invalidateFileNameNode(
+    node: FileNameNode,
+    filePath: FilePath,
+    matchNodes: Array<RequestGraphNode>,
+  ) {
+    // If there is an edge between this file_name node and one of the original file nodes pointed to
+    // by the original file_name node, and the matched node is inside the current directory, invalidate
+    // all connected requests pointed to by the file node.
+    let dirname = path.dirname(filePath);
+    for (let matchNode of matchNodes) {
+      if (
+        matchNode.type === 'file' &&
```

Comment:
Maybe filter based on the type in `respondToFSEvents` so it doesn't have to be checked for every directory?

Reply:
well... all nodes connected by `invalidated_by_create_above` edges will be file nodes, the check here was really to appease flow.

Reply:
This might be cleaner
```diff
diff --git packages/core/core/src/RequestTracker.js packages/core/core/src/RequestTracker.js
index 5f5cc30c..1468b384 100644
--- packages/core/core/src/RequestTracker.js
+++ packages/core/core/src/RequestTracker.js
@@ -488,7 +488,7 @@ export class RequestGraph extends Graph<
   invalidateFileNameNode(
     node: FileNameNode,
     filePath: FilePath,
-    matchNodes: Array<RequestGraphNode>,
+    matchNodes: Array<FileNode>,
   ) {
     // If there is an edge between this file_name node and one of the original file nodes pointed to
     // by the original file_name node, and the matched node is inside the current directory, invalidate
@@ -496,7 +496,6 @@ export class RequestGraph extends Graph<
     let dirname = path.dirname(filePath);
     for (let matchNode of matchNodes) {
       if (
-        matchNode.type === 'file' &&
         this.hasEdge(node.id, matchNode.id, 'invalidated_by_create_above') &&
         isDirectoryInside(path.dirname(matchNode.value.filePath), dirname)
       ) {
@@ -547,7 +546,10 @@ export class RequestGraph extends Graph<
           let above = this.getNodesConnectedTo(
             fileNameNode,
             'invalidated_by_create_above',
-          );
+          ).map(n => {
+            invariant(n.type === 'file');
+            return n;
+          });
           if (above.length > 0) {
             this.invalidateFileNameNode(fileNameNode, filePath, above);
           }
```